.github/pull_request_template.md

```diff
@@ -10,16 +10,22 @@ List any other tasks / PRs that are required to be merged alongside this one (e.
 
 # Checklist:
 
-These are always required, if you do not do them, reviewers will be angry:
+If any of these are not applicable strikethrough the line `~~like this~~`. **Do not delete it!**. Let the reviewer decide if you should have done it.
+
+### Code
 - [ ] I have performed a self review of my own code.
 - [ ] I have added/updated unit tests for these changes, and if not I have explained why they are not necessary.
-
-These you can remove from the list if they are not applicable for this PR.
-- [ ] I have updated the changelog.
-- [ ] I have updated any documentation required for these changes.
 - [ ] I have commented my code in any hard-to-understand or hacky areas.
 - [ ] I have handled all new warnings generated by the compiler or IDE.
```

Comment:
Not a change for now, just leaving my thoughts.
In the long term I'd like to remove this, in line with my comment [here](https://github.com/zepben/.github/pull/14#issuecomment-1769766344). I think we should ideally automate this sort of stuff, and we shouldn't need to review it. We can block merging in CI if there's warnings we don't like.

Reply:
I totally agree in principle. In reality we are a long way from this one. Because there hasn't been judicious checking in the past, certain parts of the code base are riddled with warnings and the path to having this checked and blocked by CI is going to be incremental. Especially when the warnings generated by the IDE often are not generated by the compiler.

The new lines at ends of file are probably an easier target. However, we do not have any linting for our backend code (we rely on the IDE formatter rules). So, there is definitely work there.

This PR template is not set in stone and it's easy to change. I'd rather call things out now, so people actually fix them before we have automation to do that. When we get the automation, we remove the appropriate bit of the template.

Reply:
Agree 😄 

> we do not have any linting for our backend code (we rely on the IDE formatter rules).

IntelliJ formatting can be run via CLI: https://www.jetbrains.com/help/idea/command-line-formatter.html

From a quick look, it seems possible to integrate this into our CI:
- https://github.com/marketplace/actions/intellij-format-checker
	- Any formatting fixes are left as review comments for the PR owner to accept.
- https://hub.docker.com/r/ccaominh/intellij-inspect
	- A docker container for us to manually run the formatting command and do our own work with the output.

Would something like this be an improvement to make?

Reply:
Definitely is the right path. I assume they just use the community edition, which raises a couple of questions:
- How often do they update these to get the latest IDE tools so they work the same way as what devs are actually using?
- Does this work for webstorm / front end stuff (which isn't supported by the community edition).

Worst case we have to roll our own actions / containers taking inspiration from these things. Best case is we get something basic in place using one of these things.

Please create a clickup task (probably in Devops->CI space) and the discussion can be moved there (or in slack and summary put there).

Reply:
Created https://app.clickup.com/t/86ctt0ccm 🦆